include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -178,28 +179,73 @@ struct operational_domain
      * different physical parameters, the parameters are stored with the corresponding operating status.
      */
     locked_parallel_flat_hash_map<Key, Value> operational_values{};
+    /**
+     * This can store different information depending on the use case. If the critical temperature is simulated on top
+     * of the operational domain, it is stored here.
+     */
+    locked_parallel_flat_hash_map<Key, MetricType> metric_values{};
```

Comment:
Why is it necessary to distinguish between a metric and a value? Couldn't a temperature domain just be an `operational_domain<parameter_point, double>` where `Value` is `double` and represents the critical temperature? Maybe we'd find a better name for `Value` then :wink:

Reply:
True! There is no need for that.